### PR TITLE
refactor: create BeagleScope to remove GlobalScope

### PR DESCRIPTION
**Pull request by viniciusguardieirozup**
### Description and Example

We have GlobalScope on BeagleSdk, on this PR, I create a BeagleScope to remove this GlobalScope

### Checklist

Please, check if these important points are met using `[x]`:

- [x] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [x] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [x] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

File: android/beagle/src/main/java/br/com/zup/beagle/android/utils/BeagleScope.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package br.com.zup.beagle.android.utils
+
+import br.com.zup.beagle.android.utils.CoroutineDispatchers
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/utils/BeagleScope.kt:19]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/bf1c6675d25aad5b186a17823dfcdd9be0388215/android/beagle/src/main/java/br/com/zup/beagle/android/utils/BeagleScope.kt#L19"></span>The import ‘br.com.zup.beagle.android.utils.CoroutineDispatchers’ is unused.</td>
    </tr>
  </tbody>
</table>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">The title of PR not follow the rules of: https://www.conventionalcommits.org/en/v1.0.0/</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/beagle/src/main/java/br/com/zup/beagle/android/utils/BeagleScope.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package br.com.zup.beagle.android.utils
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+
+internal class BeagleScope : CoroutineScope{
```
Comment (uziasferreirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/utils/BeagleScope.kt:22]:
```suggestion
internal class BeagleScope : CoroutineScope {
```

**Review by hthemir — DISMISSED** [review]

**Review by uziasferreirazup — APPROVED** [review]

**Review by paulomeurerzup — DISMISSED** [review]

File: android/beagle/src/main/java/br/com/zup/beagle/android/utils/BeagleScope.kt [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ */
+
+package br.com.zup.beagle.android.utils
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+
+internal class BeagleScope : CoroutineScope {
+
+    private val job = Job()
```
Comment (paulomeurerzup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/utils/BeagleScope.kt:24]: I think we should be using a SupervisorJob instead. Using a SupervisorJob, a failure or cancellation of a child does not cause the supervisor job to fail and does not affect its other children.

Let's say we've started two coroutines in BeagleScope. Without SupervisorJob if one coroutine fails, the other coroutine will be cancelled too.

File: android/beagle/src/main/java/br/com/zup/beagle/android/utils/BeagleScope.kt [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    override val coroutineContext = job + CoroutineDispatchers.IO
```
Comment (paulomeurerzup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/utils/BeagleScope.kt:25]: Should we be using CoroutineDispatchers.IO as default? This dispatcher is optimized to disk and network tasks. 

Maybe we should use CoroutineDispatchers.Main as default and let the user change the context of its own coroutines as needed.

**Review by paulomeurerzup — APPROVED** [review]

File: android/beagle/src/main/java/br/com/zup/beagle/android/utils/BeagleScope.kt [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package br.com.zup.beagle.android.utils
+
+import kotlinx.coroutines.CoroutineDispatcher
```
Comment (botbeaglezup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/utils/BeagleScope.kt:19]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/5a7378ec8de128a915c457a75ddfc8b4d09c265c/android/beagle/src/main/java/br/com/zup/beagle/android/utils/BeagleScope.kt#L19"></span>The import ‘kotlinx.coroutines.CoroutineDispatcher’ is unused.</td>
    </tr>
  </tbody>
</table>

**Review by carlossteinzup — DISMISSED** [review]

**Review by paulomeurerzup — DISMISSED** [review]

File: android/beagle/src/main/java/br/com/zup/beagle/android/utils/BeagleScope.kt (hunk 5 above)
Comment (uziasferreirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/utils/BeagleScope.kt:19]: Remove the unused import

**Review by uziasferreirazup — CHANGES_REQUESTED** [review]

**Review by uziasferreirazup — APPROVED** [review]

**Review by FredericoNoce — APPROVED** [review]

**Pull request merged**